pipeline/src/resources/Annotation.py

```diff
@@ -1,7 +1,6 @@
-
 class Annotation():
     def __iter__(self):
         return self.__dict__.items().__iter__()
 
-    def name(self):
-        return type(self).__name__
+    def name(self=None):
```

Comment:
I have mixed feelings about self as an optional parameter. Do you think it would it make more sense to do `x.__class__.name()` instead of allowing the call to either the class or instances?

Reply:
My first reaction is to agree with Lewis - `name` makes sense on the class and the instance, and should be the same for both.

Reply:
I actually prefer how Java handles situations like this. I'd like to make it a static method, which can be called from either the class or an instance of the class, but this is the only way I found to implement it in Python. `x.__class__.name()` could work, but I don't really know Python-isms enough to vote one way or the other. Getting the class still feels a bit dirty, though.

Reply:
How about this instead?

https://stackoverflow.com/a/735978/6733371

```
class TestClass():
    @staticmethod
    def test():
        print("Hello")

x = TestClass()
x.test()
TestClass.test()
```